.github/workflows/recommend-integration-tests.yml

```
@@ -0,0 +1,73 @@
+name: Recommend integration tests
+on:
+  pull_request:
+
+jobs:
+  recommend:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: 22
+
+      - name: Get source files changes
+        id: source-files
+        run: |
+          DIFF=$(git diff --name-only origin/main | grep 'packages/react' | grep -Ev '.stories.tsx|.docs.json' || true)
+          if [ -z "$DIFF" ]; then
+            echo "diff=" >> $GITHUB_OUTPUT
+          else
+            echo "diff=$DIFF" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Has diff?
+        run: echo ${{ steps.source-files.outputs.diff != '' }}
+
+      - name: Get or Create Comment
+        if: ${{ steps.source-files.outputs.diff != '' }}
+        uses: actions/github-script@v7
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            const result = await github.rest.issues.listComments({
+              issue_number: context.issue.number,
+              owner: context.repo.owner,
+              repo: context.repo.repo
+            });
```

I think it's unlikely we'd run into this but we may end up with a chain of new comments if the number of comments exceed the default page size since we would not be able to find the previous comment.

I think we could solve that with `github.paginate` on this line


```suggestion
            const result = await github.paginate(github.rest.issues.listComments, {
              issue_number: context.issue.number,
              owner: context.repo.owner,
              repo: context.repo.repo
            });
```

---

ah yes, good note!